### PR TITLE
docs: Redirect deprecated Documentation URLs to the current Github ones

### PR DESCRIPTION
**Pull request by almir-okato**
Some external MCUboot tutorials and posts still have the old URL reference for documentation and they are not up to date. 
This PR makes the redirection of these URL to the actual documentation pages.

### CODE REVIEW

**Comment by almir-okato** [comment]
@d3zd3z @kylekirkby PTAL

File: Gemfile [hunk 1]
```diff
@@ -4,4 +4,5 @@ gem "linaro-jekyll-theme", "4.3.3"
 group :jekyll_plugins do
     gem "closure-compiler"
     gem "nokogiri"
+    gem "jekyll-redirect-from"
```
Comment (d3zd3z) [inline Gemfile:7]: Missing newline at the end of the file.

**Review by d3zd3z — CHANGES_REQUESTED** [review]
Otherwise, looks good. This will help people stop getting the old docs.

**Comment by kylekirkby** [comment]
Hi @almir-okato ,

I'm sorry but we already have a mechanism in place for adding redirects to our Jekyll sites

https://github.com/mcu-tools/website-mcuboot/blob/main/_data/routingrules.json

Please add your redirects to this file.

These redirects will then be deployed via lambda@edge.

Cheers!

Kyle

**Comment by kylekirkby** [comment]
> @kylekirkby I tried to use this file, however it didn't work, and I could not find how to properly do that using the file. Could you please provide an example?

Hi @almir-okato ,

So for `/documentation/SECURITY/` (_documentation/SECURITY.md) add a line like:

```
  "^/documentation/SECURITY(/?|/index.html)$ https://docs.mcuboot.com/SECURITY  [R,L,NC]",
```

You can see the `routingrules.json` for the Linaro.org website, which is using the same website build stack, here: 

https://github.com/Linaro/website/blob/master/_data/routingrules.json

**Comment by almir-okato** [comment]
Hi @kylekirkby 
I did as you said, however the link checking job is failing, do you have any suggestions on what could be missing or wrong?
Thanks!

**Comment by kylekirkby** [comment]
Hi @almir-okato,

The failed twitter links were due to an issue with the Link Checker script / Twitter, I've re-run the failed GH jobs and the checks have passed. Apologies about this! 

Cheers,

Kyle

**Comment by almir-okato** [comment]
Hi @kylekirkby, thank you!
Would be possible to merge?

**Review by kylekirkby — APPROVED** [review]

**Review by kylekirkby — CHANGES_REQUESTED** [review]
Do we not want to keep these old documentation redirects in the file?

It may be a good idea for legacy links?

**Comment by kylekirkby** [comment]
> Do we not want to keep these old documentation redirects in the file?
> 
> It may be a good idea for legacy links?

If not, then it looks good to me :)

**Comment by almir-okato** [comment]
> Do we not want to keep these old documentation redirects in the file?
> 
> It may be a good idea for legacy links?

Did you mean that we need to remove the deprecated pages files from repo?

**Comment by kylekirkby** [comment]
> > Do we not want to keep these old documentation redirects in the file?
> > It may be a good idea for legacy links?
> 
> Did you mean that we need to remove the deprecated pages files from repo?

I mean the old redirects that you've replaced e.g:

```
  "^/release-notes.html$ /documentation/release-notes/ [R,L,NC]",
```
It might be worth keeping these and updating them to point to the new locations?

**Comment by almir-okato** [comment]
@d3zd3z could you please take a look again?
Thanks!

**Comment by almir-okato** [comment]
@d3zd3z could you please take a look again? Its pending for merging.
Thanks!

**Review by d3zd3z — APPROVED** [review]

**Pull request merged**